lib/db/set.go

```diff
@@ -196,6 +196,9 @@ type Snapshot struct {
 func (s *FileSet) Snapshot() *Snapshot {
 	t, err := s.db.newReadOnlyTransaction()
 	if err != nil {
+		if backend.IsClosed(err) {
+			return nil
```

Comment:
I expect we're just going to panic one later when we call a method on the nil snapshot instead?

Reply:
No, we don't close the db until the model is stopped and the snapshot is only used there. It's the same as with `NewFileSet` (that rabbit hole wasn't dug by me for once as far as I remember :D ).

Reply:
Well, if this method is called at all then surely its return value will be used immediately thereafter?

Reply:
Oh damn, so the panic shouldn't happen in the first place. I retract my statement and go back to thinking.

Reply:
And *cough* #5671 *cough*